src/app/model/ChessEngine.ts

```diff
@@ -119,4 +138,44 @@ export class ChessEngine implements ChessModel {
   public possibleMoves(position: Square): Square[] {
     return this.pieceMovesNotCausingAllyKingCheckmate(position);
   }
+
+  private onKingWasChecked(event: KingWasChecked): void {
+    this.checkedKing = { kingSide: event.king.side, position: event.onSquare };
+  }
+
+  private kingWasChecked(chosenPiece: Piece): KingWasChecked | undefined {
+    const kingsSide = chosenPiece.side === Side.WHITE ? Side.BLACK : Side.WHITE;
+    const kingPosition = this.kingPosition(this.board, kingsSide);
+    if (!kingPosition) {
+      return undefined;
+    }
+    const isKingChecked = this.isKingChecked(this.board, kingsSide);
+    return isKingChecked
+      ? {
+          eventType: 'KingWasChecked',
+          king: new King(kingsSide),
+          onSquare: kingPosition,
+        }
+      : undefined;
+  }
+
+  private kingWasUnchecked(pieceWasMoved: PieceWasMoved): KingWasUnchecked | undefined {
+    if (!this.checkedKing) {
+      return undefined;
+    }
+    const kingSide =
```

Comment:
Wowow, co się tutaj narobiło :P Przecież to masz zapisane w `checkedKing.side`

Reply:
hahahah dobra nawet nic nie mów, w sensie dzięki temu zaczęło mi to działać, ale musiałem to przegadać z Anią i takie coś nam powstało xD teraz nastała jasność